MvvmCross.Forms/Platform/Android/Views/MvxFormsAppCompatActivity.cs

```diff
@@ -93,26 +94,43 @@ protected override void AttachBaseContext(Context @base)
             base.AttachBaseContext(MvxContextWrapper.Wrap(@base, this));
         }
 
-        protected override void OnCreate(Bundle bundle)
+        protected override async void OnCreate(Bundle bundle)
         {
             // Required for proper Push notifications handling      
             var setupSingleton = MvxAndroidSetupSingleton.EnsureSingletonAvailable(ApplicationContext);
             setupSingleton.EnsureInitialized();
 
             base.OnCreate(bundle);
+
+            await InternalOnCreate(bundle);
+        }
+
+        protected virtual async Task InternalOnCreate(Bundle bundle)
+        {
             ViewModel?.ViewCreated();
+
+            await StartSetup();
+
             InitializeForms(bundle);
         }
 
-        public virtual void InitializeForms(Bundle bundle)
+        protected virtual async Task StartSetup()
```

Comment:
Maybe this method should be called something like `RunAppStart`, the `Setup` keyword is a bit confusing in this context

Reply:
Good call. Will fix